### PR TITLE
[batch] Use a compound key for the job id

### PR DESCRIPTION
**Pull request by jigold**
This PR is needed for atomic batch creation where the job_id is 0 to n_jobs and is not unique across batches.

- Job now has a compound key in the database (batch_id, job_id)
- Job.id on the server side is `(batch_id, job_id)`
- Batch on the client side has been modified to not require the attributes since Job now needs a reference to the batch.
- Changed some routes to be '/batches/{batch_id}/jobs/{job_id}/...`
- Changed the appropriate places in ci2 and pipeline that use the job status interface
- Changed the create_job interface for parents to take the Job objects and not the id. I think this is clearer and will allow me to check the batch of the Job is the correct batch in the future.
- deleted creating batch from a file

### CODE REVIEW

File: batch/create-batch-tables.sql [hunk 1]
```diff
@@ -38,20 +38,21 @@ CREATE TABLE IF NOT EXISTS `jobs` (
   `input_log_uri` VARCHAR(1024),
   `main_log_uri` VARCHAR(1024),
   `output_log_uri` VARCHAR(1024),
-  PRIMARY KEY (`id`),
+  PRIMARY KEY (`batch_id`, `job_id`),
   FOREIGN KEY (`batch_id`) REFERENCES batch(id) ON DELETE CASCADE
 ) ENGINE = InnoDB;
 CREATE INDEX jobs_user ON jobs (user);
 CREATE INDEX jobs_batch ON jobs (batch_id);
```
Comment (danking) [inline batch/create-batch-tables.sql]: this index is completely superseded, AFIAK, by the primary key. It's possible I don't understand postgres well, but usually the primary key defines a b-tree on the data in a way that allows comparison using any prefix of the key.

File: batch/create-batch-tables.sql [hunk 2]
```diff
@@ -38,20 +38,21 @@ CREATE TABLE IF NOT EXISTS `jobs` (
...
-  FOREIGN KEY (`job_id`) REFERENCES jobs(id) ON DELETE CASCADE,
-  FOREIGN KEY (`parent_id`) REFERENCES jobs(id) ON DELETE CASCADE
+  `batch_id` BIGINT NOT NULL,
+  `job_id` INT NOT NULL,
+  `parent_id` INT NOT NULL,
+  PRIMARY KEY (`batch_id`, `job_id`, `parent_id`),
+  FOREIGN KEY (`batch_id`) REFERENCES batch(id) ON DELETE CASCADE
 ) ENGINE = InnoDB;
-CREATE INDEX jobs_parents_parent_id ON `jobs-parents` (parent_id);
+CREATE INDEX jobs_parents_job_id ON `jobs-parents` (batch_id, job_id);
```
Comment (danking) [inline batch/create-batch-tables.sql]: likewise, I believe this index is unnecessary (and uses disk/mem space).

**Comment by danking** [comment]
Assigning @akotlar for shepherding as I'll be away the next three days.

**Comment by akotlar** [comment]
I'll be able to take a closer look after 5pm tomorrow, or after Dan's comments addressed

**Comment by jigold** [comment]
@akotlar No rush. This rebase looks like it's going to be ugly and should be stacked on #6208

**Comment by jigold** [comment]
It was too difficult to split this functionality apart from the larger change. I am going to work on #6230 instead.

**Comment by akotlar** [comment]
Sounds good!

**Pull request closed without merging**